integration-libs/opf/base/core/services/opf-payment-hosted-fields.service.ts

```diff
@@ -0,0 +1,203 @@
+/*
+ * SPDX-FileCopyrightText: 2023 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import { ActiveCartFacade } from '@spartacus/cart/base/root';
+import {
+  CommandService,
+  GlobalMessageService,
+  RoutingService,
+  UserIdService,
+  WindowRef,
+} from '@spartacus/core';
+import { Order } from '@spartacus/order/root';
+
+import { EMPTY, Observable, combineLatest, from, throwError } from 'rxjs';
+import {
+  catchError,
+  concatMap,
+  filter,
+  map,
+  switchMap,
+  take,
+  tap,
+} from 'rxjs/operators';
+import { OpfOrderFacade, OpfOtpFacade } from '../../root/facade';
+import {
+  MerchantCallback,
+  OpfPaymentError,
+  PaymentErrorType,
+  PaymentMethod,
+  SubmitCompleteInput,
+  SubmitCompleteRequest,
+  SubmitCompleteResponse,
+  SubmitInput,
+  SubmitRequest,
+  SubmitResponse,
+  SubmitStatus,
+  defaultError,
+} from '../../root/model';
+import { OpfPaymentConnector } from '../connectors/opf-payment.connector';
+import { OpfPaymentErrorHandlerService } from '../services/opf-payment-error-handler.service';
+import { getBrowserInfo } from '../utils/opf-payment-utils';
+
+@Injectable({ providedIn: 'root' })
+export class OpfPaymentHostedFieldsService {
+  constructor(
+    protected commandService: CommandService,
+    protected opfPaymentConnector: OpfPaymentConnector,
+    protected winRef: WindowRef,
+    protected opfOtpFacade: OpfOtpFacade,
+    protected activeCartFacade: ActiveCartFacade,
+    protected userIdService: UserIdService,
+    protected routingService: RoutingService,
+    protected opfOrderFacade: OpfOrderFacade,
+    protected globalMessageService: GlobalMessageService,
+    protected opfPaymentErrorHandlerService: OpfPaymentErrorHandlerService
+  ) {}
+
+  submitPayment(submitInput: SubmitInput): Observable<boolean> {
```

Comment:
this method (submitPayment) was just moved from opf-payment.service to opf-payment-hosted-fields.service, so has been reviewd when pattern#1 was implemented